plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/TrinoNessieCatalogFactory.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.nessie;
+
+import io.trino.plugin.base.CatalogName;
+import io.trino.plugin.hive.NodeVersion;
+import io.trino.plugin.iceberg.IcebergConfig;
+import io.trino.plugin.iceberg.catalog.IcebergTableOperationsProvider;
+import io.trino.plugin.iceberg.catalog.TrinoCatalog;
+import io.trino.plugin.iceberg.catalog.TrinoCatalogFactory;
+import io.trino.spi.security.ConnectorIdentity;
+import io.trino.spi.type.TypeManager;
+
+import javax.inject.Inject;
+
+import static java.util.Objects.requireNonNull;
+
+public class TrinoNessieCatalogFactory
+        implements TrinoCatalogFactory
+{
+    private final IcebergTableOperationsProvider tableOperationsProvider;
+    private final String warehouseLocation;
+    private final NessieIcebergClient nessieClient;
+    private final boolean isUniqueTableLocation;
+    private final String trinoVersion;
+    private final CatalogName catalogName;
+    private final TypeManager typeManager;
+
+    @Inject
+    public TrinoNessieCatalogFactory(
+            CatalogName catalogName,
+            TypeManager typeManager,
+            IcebergTableOperationsProvider tableOperationsProvider,
+            NessieIcebergClient nessieClient,
```

Comment:
Can we abstain from injecting the client and create it on the fly instead?

Reply:
I think we can, but I would probably leave these changes until we can re-use a bit more code around the NessieIcebergClient from Iceberg directly